### PR TITLE
New barman stack

### PR DESCRIPTION
**Pull request by ClementGuittat**
PR concernant les 5 premières requêtes de SwaggerHub. 

- [x] PR is rebased on top of `origin/master` 
- [x] PR provide new tests for new behaviors
- [x] PR doesn't introduce commented out code
- [ ] PR has been tested locally on the main use cases
- [x] Documentation have been updated if necessary

### CODE REVIEW

File: server/app/controllers/barman-controller.js [hunk 1]
```diff
@@ -13,6 +15,126 @@ async function getAllBarmen(req, res) {
...
+
+/**
+* Upadate a barman.
+* @param req Request
+* @param res Response
+* @return {Promise.<void>} Nothing
+*/
+async function updateBarman(req, res) {
+    // FIXME We should check the type of each provided field, instead of just the presence
+    if (!checkStructure(req.body, ['firstName', 'lastName'])) {
```
Comment (ClementGuittat) [inline server/app/controllers/barman-controller.js]: je suis pas sur que j'ai besoin seulement du nom et prénom

File: server/app/services/barman-service.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 const logger = require('../../logger');
 const { Barman } = require('../models/barman');
+//const { Service } = require ('../models/service');
```
Comment (kistora) [inline server/app/services/barman-service.js]: Not used for now, you schould remove it

File: server/app/routes/barmen.js [hunk 3]
```diff
@@ -5,4 +5,14 @@ const barmanController = require('../controllers/barman-controller');
 
 router.get('/', guard.check('barman:read'), am(barmanController.getAllBarmen));
 
+router.post('/', guard.check('barman:write'), am(barmanController.createBarman));
+
+router.get('/:id(\\d+)', guard.check('barman:read'), am(barmanController.getBarmanById));
+router.put('/:id(\\d+)', guard.check('barman:write'), am(barmanController.updateBarman));
+router.delete('/:id(\\d+)', guard.check('barman:write'), am(barmanController.deleteBarman));
+
+//router.get('/:id(\\d+)/services', guard.check('barman:read, service:read'), am(barmanController.getServicesBarman));
```
Comment (kistora) [inline server/app/routes/barmen.js]: Not used for now, remove it :) Same comment for the lines bellow

File: server/app/services/barman-service.js [hunk 4]
```diff
@@ -8,10 +10,118 @@ const { Barman } = require('../models/barman');
...
+
+    return await currentBarman.update({
+        firstName: updatedBarman.firstName,
+        lastName: updatedBarman.lastName,
+        nickname: updatedBarman.nickname,
+        facebook: updatedBarman.facebook,
+        godFather: updatedBarman.godFather,
+        dateOfBirth: updatedBarman.dateOfBirth,
+        flow: updatedBarman.flow,
+        kommissions: updatedBarman.kommissions,
```
Comment (kistora) [inline server/app/services/barman-service.js]: Do not update the association fields for now :) So remove kommissions, roles & services line

File: server/app/controllers/barman-controller.js [hunk 5]
```diff
@@ -13,6 +15,126 @@ async function getAllBarmen(req, res) {
...
+            'BadRequest',
+            'The body has missing properties, needed: [\'firstName\', \'lastName\']'
+        );
+    }
+
+    let newBarman = new Barman({
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        nickname: req.body.nickname,
+        facebook: req.body.facebook,
```
[the remark this answers is not in the document]
Comment (Embraser01) [reply server/app/controllers/barman-controller.js]: I didn't use it before neither, do you think we should add it (just for readability) ?

**Pull request merged**